targets/linux/distro_handler.go

```diff
@@ -90,30 +90,66 @@ func HandleContainer(c DistroConfig) gwclient.BuildFunc {
 				return nil, nil, err
 			}
 
-			pg := dalec.ProgressGroup(spec.Name)
-			pc := dalec.Platform(platform)
+			var opts []llb.ConstraintsOpt
+			opts = append(opts, dalec.ProgressGroup(spec.Name))
+			opts = append(opts, dalec.Platform(platform))
 
-			worker, err := c.Worker(sOpt, pg, pc)
+			worker, err := c.Worker(sOpt, opts...)
 			if err != nil {
 				return nil, nil, err
 			}
-
-			deb, err := c.BuildPkg(ctx, client, worker, sOpt, spec, targetKey, pg, pc)
+			// We need to get the build context from the client.
+			inputs, err := client.Inputs(ctx)
 			if err != nil {
-				return nil, nil, err
+				return nil, nil, fmt.Errorf("error getting llb state from client with provided build context: %w", err)
+			}
+
+			// This makes sure it obeys include/exclude filters when called later on by the image resolver logic.
+			sOpt.GetContext = func(name string, opts ...llb.LocalOption) (*llb.State, error) {
```

Comment:
This should already be set on `sOpt`, no need to change anything.
Should definitely change this to an interface at some point rather than a struct with closures.

Reply:
What we want is to not look at `inputs` lower down directly and instead just call `sOpt.GetContext(<name>, opts...)`.
As mentioned above, `sOpt` should already be fully setup.

Reply:
The opts we have here is of type:
```
type ConstraintsOpt interface {
	SetConstraintsOption(*Constraints)
	RunOption
	LocalOption
	HTTPOption
	ImageOption
	GitOption
	OCILayoutOption
}
```

And the one that sOpt.GetContext expects is:
```
type LocalOption interface {
	SetLocalOption(*LocalInfo)
}
```

Should I be initializing a diff opts separate from the one that we're working with in the function or is there a way to easily translate it over to the local version?

Reply:
All constraints opts implement LocalOption, no need to filter them out.

Reply:
You can do `sOpt.GetContext(<name>, WithConstraints(opts...))`. `WithConstraintsOpts` combines multiple constraints opts into one, and that that constraints opt can be downcast into a `LocalOption`